doc/developer/lua.rst

```diff
@@ -0,0 +1,59 @@
+LUA
+===
+
+LUA is currently experimental within FRR and has very limited
+support.  If you would like to compile FRR with LUA you must
+follow these steps
+
+1. Installation of Relevant Libraries
+
+   .. code-block:: console
+       apt-get install lua5.3 liblua5-3 liblua5.3-dev
+
+   These are the Debian libraries that are needed.  There should
+   be equivalent RPM's that can be found
+
+2. Compilation
+
+   Configure needs these options
+
+   .. code-block:: console
+       ./configure --enable-dev-build --enable-lua <all other interesting options>
+
+   Typically you just include the two new enable lines to build with it.
+
+3. Using Lua
+
+   * Copy tools/lua.scr into /etc/frr
+
+   * Create a route-map match command
+
+   .. code-block:: console
+      !
+      router bgp 55
+        neighbor 10.50.11.116 remote-as external
+        address-family ipv4 unicast
+          neighbor 10.50.11.116 route-map TEST in
+        exit-address-family
+      !
+      route-map TEST permit 10
+        match command mooey
+      !
+
+   * In the lua.scr file make sure that you hae a function named 'mooey'
+
+   .. code-block:: console
+       function mooey ()
+           zlog_debug(string.format("afi: %d: %s %d ifdx: %d aspath: %s localpref: %d",
+                      prefix.family, prefix.route, nexthop.metric,
+                      nexthop.ifindex, nexthop.aspath, nexthop.localpref))
+    
+           nexthop.metric =  33
+           nexthop.localpref = 13
+           return 3
```

Comment:
what does return 3 mean?

Reply:
they need to read the code ;) at this point.